browser/js/common/directives/navbar/navbar.js

```diff
@@ -1,47 +1,71 @@
-app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state) {
+app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state, $window) {
 
-    return {
-        restrict: 'E',
-        scope: {},
-        templateUrl: 'js/common/directives/navbar/navbar.html',
+	return {
+		restrict: 'E',
+		scope: {},
+		templateUrl: 'js/common/directives/navbar/navbar.html',
 
-        link: function (scope) {
+		link: function (scope) {
+			var temp = 0;
+			scope.goToSection = function(val, state) {
+				if (val !== 'null') {
+					Promise.resolve()
+					.then( () => {
+						if (!state) {
+							state = 'home';
+						}
+						if($state.current.name === state) {
+							return;
+						} else {
+						return $state.go(state);
+						}
+					})
+					.then( () => {
+						// temp = $window.pageYOffset;
+						// $window.pageYOffset -= 58;
+						smoothScroll(document.querySelector(val));
+					});
```

Comment:
realistically, we should `.catch` off this promise chain as it is _possible_ that the state-change promise could be rejected, though i'm not sure what we would want to happen in terms of the UX if it were to fail. i'll add something for it later.